From 9e03995b0ae270eb8901efaca248cd685de772cf Mon Sep 17 00:00:00 2001
From: Stefan Strogin <steils@gentoo.org>
Date: Tue, 31 Dec 2019 03:22:26 +0200
Subject: [PATCH] Call the correct method for creating TreeModelSort (Closes
 #2554)

---
 lutris/gui/views/store.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lutris/gui/views/store.py b/lutris/gui/views/store.py
index 9bab31587..9e843e9e4 100644
--- a/lutris/gui/views/store.py
+++ b/lutris/gui/views/store.py
@@ -140,7 +140,7 @@ def __init__(
         self.prevent_sort_update = False  # prevent recursion with signals
         self.modelfilter = self.store.filter_new()
         self.modelfilter.set_visible_func(self.filter_view)
-        self.modelsort = Gtk.TreeModelSort.sort_new_with_model(self.modelfilter)
+        self.modelsort = Gtk.TreeModelSort.new_with_model(self.modelfilter)
         self.modelsort.connect("sort-column-changed", self.on_sort_column_changed)
         self.modelsort.set_sort_func(sort_col, sort_func, sort_col)
         self.sort_view(sort_key, sort_ascending)
