From 05d930148a47ed97dd0afb1b0a91045f85d87bdd Mon Sep 17 00:00:00 2001
From: gasinvein <gasinvein@gmail.com>
Date: Tue, 10 Sep 2019 13:01:07 +0300
Subject: [PATCH 1/3] Add Flatpak-specific Vulkan ICD paths

---
 lutris/sysoptions.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lutris/sysoptions.py b/lutris/sysoptions.py
index 36278ae0..a9865ecb 100644
--- a/lutris/sysoptions.py
+++ b/lutris/sysoptions.py
@@ -23,6 +23,8 @@ def get_optirun_choices():
 def get_vk_icd_choices():
     """Return available Vulkan ICD loaders"""
     loader_paths = ["/usr/share/vulkan/icd.d/*.json",  # standard location
+                    "/usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/*.json", # Flatpak GL extension
+                    "/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/*.json", # Flatpak GL32 extension
                     "/opt/amdgpu-pro/etc/vulkan/icd.d/*.json",  # AMD GPU Pro - TkG
                     "/etc/vulkan/icd.d/*.json"]  # AMDVLK - Ubuntu
     choices = [("Auto", "")]

From ed23d83faa8cab159acaa1c826660e56e4b8c6da Mon Sep 17 00:00:00 2001
From: gasinvein <gasinvein@gmail.com>
Date: Tue, 10 Sep 2019 13:47:20 +0300
Subject: [PATCH 2/3] Add more Vulkan data dirs Put them into a separate
 constant and use Vulkan-spec-compliant order

---
 lutris/sysoptions.py | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/lutris/sysoptions.py b/lutris/sysoptions.py
index a9865ecb..be841dee 100644
--- a/lutris/sysoptions.py
+++ b/lutris/sysoptions.py
@@ -8,6 +8,17 @@
 from lutris.discord import DiscordPresence
 
 
+VULKAN_DATA_DIRS = [
+    "/usr/local/etc/vulkan",  # standard site-local location
+    "/usr/local/share/vulkan",  # standard site-local location
+    "/etc/vulkan",  # standard location
+    "/usr/share/vulkan",  # standard location
+    "/usr/lib/x86_64-linux-gnu/GL/vulkan",  # Flatpak GL extension
+    "/usr/lib/i386-linux-gnu/GL/vulkan",  # Flatpak GL32 extension
+    "/opt/amdgpu-pro/etc/vulkan"  # AMD GPU Pro - TkG
+]
+
+
 def get_optirun_choices():
     """Return menu choices (label, value) for Optimus"""
     choices = [("Off", "off")]
@@ -22,11 +33,9 @@ def get_optirun_choices():
 
 def get_vk_icd_choices():
     """Return available Vulkan ICD loaders"""
-    loader_paths = ["/usr/share/vulkan/icd.d/*.json",  # standard location
-                    "/usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/*.json", # Flatpak GL extension
-                    "/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/*.json", # Flatpak GL32 extension
-                    "/opt/amdgpu-pro/etc/vulkan/icd.d/*.json",  # AMD GPU Pro - TkG
-                    "/etc/vulkan/icd.d/*.json"]  # AMDVLK - Ubuntu
+    loader_paths = []
+    for data_dir in VULKAN_DATA_DIRS:
+        loader_paths.append(os.path.join(data_dir, "icd.d", "*.json"))
     choices = [("Auto", "")]
 
     # Add loaders

From 279781d5f73b2cdb1aab128cb7d4fa25664f8700 Mon Sep 17 00:00:00 2001
From: gasinvein <gasinvein@gmail.com>
Date: Tue, 10 Sep 2019 14:24:32 +0300
Subject: [PATCH 3/3] Simplify get_vk_icd_choices() function

---
 lutris/sysoptions.py | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/lutris/sysoptions.py b/lutris/sysoptions.py
index be841dee..08af2862 100644
--- a/lutris/sysoptions.py
+++ b/lutris/sysoptions.py
@@ -33,13 +33,11 @@ def get_optirun_choices():
 
 def get_vk_icd_choices():
     """Return available Vulkan ICD loaders"""
-    loader_paths = []
-    for data_dir in VULKAN_DATA_DIRS:
-        loader_paths.append(os.path.join(data_dir, "icd.d", "*.json"))
     choices = [("Auto", "")]
 
     # Add loaders
-    for path in loader_paths:
+    for data_dir in VULKAN_DATA_DIRS:
+        path = os.path.join(data_dir, "icd.d", "*.json")
         for loader in glob.glob(path):
             choices.append((os.path.basename(loader), loader))
 
